src/app/data/components/data-browser/data-browser.component.ts

```diff
@@ -1127,6 +1144,18 @@ export class DataBrowserComponent implements OnInit, OnDestroy {
     } else {
       this.extendedRecords.recordList = parsedRecordData;
     }
+
+    const currentRecordIds = parsedRecordData?.map((record) => record.id);
+    if (!currentRecordIds || currentRecordIds.length == 0) return;
+
+    this.projectApolloService.getRecordComments(this.projectId, currentRecordIds).pipe(first()).subscribe((comments) => {
+      if (!comments) return;
+      this.recordComments = JSON.parse(comments).reduce(function (r, a) {
+        r[a.record_id] = r[a.record_id] || [];
+        r[a.record_id].push(a);
+        return r;
+      }, Object.create(null));
+    });
```

Comment:
i'd put this request in the comment class as well so here we would only see something like 

this.commentsFilter.collectRecordComments()

or something similar.

Reply:
also I'm not super sure about the logic of the function. ill take a close look if I can pinpoint what my issues are :D

Reply:
found it, it simply doesn'T work :D 
test case: comment on record id 0 & on record id > 20 (so scrolling down will collect another batch)

Only the most recent comment is available so first comment from record id 0 then after scrolling only from id > 20

Reply:
i'll look for a fix

Reply:
This worked great for me

```suggestion
    const currentRecordIds = parsedRecordData?.map((record) => record.id);
    if (!currentRecordIds || currentRecordIds.length == 0) return;

    this.projectApolloService.getRecordComments(this.projectId, currentRecordIds).pipe(first()).subscribe((comments) => {
           if (!comments) return;
          const commentsParsed = tryParseJSON(comments);
          if (!commentsParsed) return;
          commentsParsed.forEach(e => {
            if (!this.recordComments[e.record_id]) this.recordComments[e.record_id] = [];
            this.recordComments[e.record_id].push(e);
          });
    });
```

Reply:
or not! sec still an error here :D

Reply:
missed to clear the dict if we don't extend :) 

``` Javascript
if (extend) {
      this.extendedRecords.recordList = [
        ...this.extendedRecords.recordList,
        ...parsedRecordData,
      ];
    } else {
      this.extendedRecords.recordList = parsedRecordData;
      this.recordComments = {};  //<--- this one
    }

```